src/Microsoft.ML.AutoML/Experiment/Runners/CrossValSummaryRunner.cs

```diff
@@ -70,27 +71,102 @@ internal class CrossValSummaryRunner<TMetrics> : IRunner<RunDetail<TMetrics>>
 
             // Build result objects
-            var suggestedPipelineRunDetail = new SuggestedPipelineRunDetail<TMetrics>(pipeline, avgScore, allRunsSucceeded, metricsClosestToAvg, bestModel, null);
+            var suggestedPipelineRunDetail = new SuggestedPipelineRunDetail<TMetrics>(pipeline, avgScore, allRunsSucceeded, avgMetrics, bestModel, null);
             var runDetail = suggestedPipelineRunDetail.ToIterationResult(_preFeaturizer);
             return (suggestedPipelineRunDetail, runDetail);
         }
 
+        private static TMetrics GetAverageMetrics(IEnumerable<TMetrics> metrics)
+        {
+            if (typeof(TMetrics) == typeof(BinaryClassificationMetrics))
+            {
+                var newMetrics = metrics.Select(x => x as BinaryClassificationMetrics);
+                Contracts.Assert(newMetrics != null);
+
+                var result = new BinaryClassificationMetrics(
+                    auc: GetAverageOfNonNaNScores(newMetrics.Select(x => x.AreaUnderRocCurve)),
+                    accuracy: GetAverageOfNonNaNScores(newMetrics.Select(x => x.Accuracy)),
+                    positivePrecision: GetAverageOfNonNaNScores(newMetrics.Select(x => x.PositivePrecision)),
+                    positiveRecall: GetAverageOfNonNaNScores(newMetrics.Select(x => x.PositiveRecall)),
+                    negativePrecision: GetAverageOfNonNaNScores(newMetrics.Select(x => x.NegativePrecision)),
+                    negativeRecall: GetAverageOfNonNaNScores(newMetrics.Select(x => x.NegativeRecall)),
+                    f1Score: GetAverageOfNonNaNScores(newMetrics.Select(x => x.F1Score)),
+                    auprc: GetAverageOfNonNaNScores(newMetrics.Select(x => x.AreaUnderPrecisionRecallCurve)));
+                return result as TMetrics;
+            }
+
+            if (typeof(TMetrics) == typeof(MulticlassClassificationMetrics))
+            {
+                var newMetrics = metrics.Select(x => x as MulticlassClassificationMetrics);
+                Contracts.Assert(newMetrics != null);
+
+                var result = new MulticlassClassificationMetrics(
+                    accuracyMicro: GetAverageOfNonNaNScores(newMetrics.Select(x => x.MicroAccuracy)),
+                    accuracyMacro: GetAverageOfNonNaNScores(newMetrics.Select(x => x.MacroAccuracy)),
+                    logLoss: GetAverageOfNonNaNScores(newMetrics.Select(x => x.LogLoss)),
+                    logLossReduction: GetAverageOfNonNaNScores(newMetrics.Select(x => x.LogLossReduction)),
+                    topKPredictionCount: newMetrics.ElementAt(0).TopKPredictionCount,
+                    topKAccuracy: GetAverageOfNonNaNScores(newMetrics.Select(x => x.TopKAccuracy)),
+                    // TODO:
+                    //   Figure out whether class label ordering can be different across different folds.
+                    //   If yes, whether it is possible to get the information from the objects available.
+                    perClassLogLoss: null);
```

Comment:
The classes can be different in ordering and some can be missing. The missing arises from small datasets or skewed labels where all given classes may not be represented in the training dataset. 

I'm not sure what our log-loss code does for labels which are not in the test dataset (perhaps `0.0` or `NaN`).

Reply:
Would it be possible to return the closest to average fold's `perClassLogLoss`?

Reply:
Thanks. Current method looks good for the moment. 

For averaging the `ConfusionMatrix` in the future, it stores the class names within itself. The class names can be used to align the classes and allow for proper averaging.

https://github.com/dotnet/machinelearning/blob/770327a2074d8ab39a023bc9252a713a3afa8293/src/Microsoft.ML.Data/Evaluators/Metrics/ConfusionMatrix.cs#L62

It mentions the class index may also be stored, though I'm unsure how that could arise. Perhaps for binary classification there's no class names to store?